### DISCUSSION TITLE
Cesium

### DISCUSSION THREAD

**Discussion by Paolo-lab** [iTwin/itwinjs-core#6183] [Q&A] [answered]
Hi everyone, my client asks me to visualize his artifacts using cesium and its markers.

I did the "npm i cesium" installation
 
but I get a compilation error because it can't resolve dependencies
url , browserify-zlib , https-browserify and stream-http , I attach an image of the compilation.

<img width="317" alt="errorCesium" src="https://github.com/iTwin/itwinjs-core/assets/57446652/409415b7-f332-4ce0-8553-9b82995168f3">

how can I solve it?

thank you for your precious help
Paul

**Comment by aruniverse** [comment, marked as answer]
The [cesium](https://www.npmjs.com/package/cesium) package you're pulling in is using Node.js APIs ([url](https://nodejs.org/api/url.html), [zlib](https://nodejs.org/api/zlib.html), etc) which are not available in a browser context when using [webpack 5](https://gist.github.com/ef4/d2cf5672a93cf241fd47c020b9b3066a). You will need to either provide polyfills for these yourself (refer to the last link), downgrade to Webpack 4, or patch (provide a PR) to the Cesium library to not use these node apis.
